lldb/source/Plugins/SymbolFile/DWARF/DebugNamesDWARFIndex.cpp

```diff
@@ -444,6 +503,73 @@ void DebugNamesDWARFIndex::GetNamespaces(
   m_fallback.GetNamespaces(name, callback);
 }
 
+llvm::SmallVector<CompilerContext>
+DebugNamesDWARFIndex::GetTypeQueryParentContexts(TypeQuery &query) {
+  std::vector<lldb_private::CompilerContext> &query_decl_context =
+      query.GetContextRef();
+  llvm::SmallVector<CompilerContext> parent_contexts;
+  if (!query_decl_context.empty()) {
+    auto rbegin = query_decl_context.rbegin(), rend = query_decl_context.rend();
+    // Reverse the query decl context to match parent chain.
+    // Skip the last entry, it is the type we are looking for.
+    for (auto rit = rbegin + 1; rit != rend; ++rit)
+      // Skip any context without name because .debug_names might not encode
+      // them. (e.g. annonymous namespace)
+      if ((rit->kind & CompilerContextKind::AnyType) !=
+              CompilerContextKind::Invalid &&
+          !rit->name.IsEmpty())
+        parent_contexts.push_back(*rit);
+  }
+  return parent_contexts;
+}
```

Comment:
```suggestion
  llvm::ArrayRef<lldb_private::CompilerContext> query_decl_context =
      query.GetContextRef();
  llvm::SmallVector<CompilerContext> parent_contexts;
  if (!query_decl_context.empty()) {
    // Reverse the query decl context to match parent chain.
    // Skip the last entry, it is the type we are looking for.
    for (CompilerContext ctx: llvm::reverse(query_decl_context.drop_back())) {
      // Skip any context without name because .debug_names might not encode
      // them. (e.g. annonymous namespace)
      if ((ctx.kind & CompilerContextKind::AnyType) !=
              CompilerContextKind() &&
          !ctx.name.IsEmpty())
        parent_contexts.push_back(ctx);
  }
  return parent_contexts;
}
```